### DISCUSSION TITLE
Add support for Amazon Bedrock prompt caching

### DISCUSSION THREAD

**Discussion by IsaacZhangg** [cline/cline#872] [Feature Requests]
Was just added to Amazon Bedrock on December 4th. Please add this feature. Relevent website is linked. 

https://aws.amazon.com/about-aws/whats-new/2024/12/amazon-bedrock-preview-prompt-caching/
https://aws.amazon.com/blogs/aws/reduce-costs-and-latency-with-amazon-bedrock-intelligent-prompt-routing-and-prompt-caching-preview/

**Comment by sreimchen** [comment]
I wanted to post the same here. I hope and guess it's not that much work to get that running because it is already in use for Antrophic's API in Cline, isnt it? Probably just some kind of flag. I would very much appreciate this feature because the prompting costs are very high currently and the reduction of cost by 60% to 90% is just incredible.

**Comment by kenzomartins** [comment]
This would be great to have, the token costs can add up pretty quickly.

**Comment by brianwthomas** [comment]
Plus 1 to this.  The cost using bedrock is really high due to no caching.

**Comment by chriswwhite** [comment]
Would be a great feature to add

**Comment by shrijayan** [comment]
https://github.com/cline/cline/discussions/1208

In AWS Bedrock, they support prompt caching through Prompt Routers, which have implemented caching for Anthropic and Meta models. They provide the model ARN ID to access the models. By implementing **Accept Model ARN ID in AWS Bedrock Model Selection #1208**, where we can input the model ARN ID in AWS Bedrock, this feature will be highly useful.

**Comment by brianwthomas** [comment]
This can probably be closed now that 3.7 is out and it supports caching.

**Reply by ajscriv** [reply]
@brianwthomas  does it support caching via Amazon bedrock? Can you link to the source for that?

**Reply by brianwthomas** [reply]
Open cline in VS code, go to bedrock, select Claude 3.7 and you'll see caching is enabled by default.

**Reply by brianwthomas** [reply]
I did some digging in PRs and found this: https://github.com/cline/cline/pull/1929

**Comment by brianwthomas** [comment]
Though while I say this, it doesn't seem like it's caching? Or if it is I can't tell. The cost isn't going down like if I use Anthropic directly.    If I use bedrock, the cost is about 8c per transaction whereas when I used anthropic directly I was able to get some of the requests down to a fraction of a cent.

**Reply by brianwthomas** [reply]
Looks like I don't see any caching...

![image](https://github.com/user-attachments/assets/b8c40176-bced-4fcd-8d68-f2ba1f5f6e97)

All items from today have 0 caching when It's reviewed the same file multiple times.

**Reply by djaym7** [reply]
+1, cache memory size remains 0 all the time

**Comment by GetSignal** [comment]
According to AWS tech support, only 3.5 supports caching (still in beta) and must be manually approved by AWS before it can be used. That means you can't just enable it via Cline and expect it to work. 3.7 does NOT support caching at all.
